graph-manager-docs/source/schema-reporting-protocol.mdx

```diff
@@ -0,0 +1,159 @@
+<ObtainGraphApiKey />
+
+## Protocol sequence
+
+The following sections illustrate the communication sequence between a GraphQL server (hereafter referred to as the **edge server**) and the schema reporting endpoint (**Apollo**):
+
+### Diagram
+
+```mermaid
+sequenceDiagram
+    participant Edge Server
+    participant Apollo
+    Edge Server->>Apollo: reportServerInfo(EdgeServerInfo(…) executableSchema = false)
+    alt The specified schema ID is already registered
+        Apollo->>Edge Server: Response(withExecutableSchema = false, inSeconds = 57s)
+        Note over Edge Server: Wait 57 seconds
+        Note over Edge Server: Go to top
+    else The specified schema ID is NOT already registered
+        Apollo->>Edge Server: Response(withExecutableSchema = true, inSeconds = 0s)
+        Note over Edge Server: Wait 0 seconds
+        Edge Server->>Apollo: reportServerInfo(EdgeServerInfo(…) executableSchema = '…')
+        Note over Apollo: Persist schema
+        Apollo->>Edge Server: Response(withExecutableSchema = false, inSeconds = 62s)
+        Note over Edge Server: Wait 62 seconds
+        Note over Edge Server: Go to top
+    else The reportServerInfo request failed
+        Note over Edge Server: Wait 20 seconds
+        Note over Edge Server: Go to top
+    end
+```
+
+### Step-by-step description
+
+1. On startup, the edge server executes the `service.reportServerInfo` mutation, providing an `EdgeServerInfo` object with the server's details as input.
+    * This initial call **does not** include an `executableSchema` string.
+    * This initial call **does** include an `executableSchemaId` in the `EdgeServerInfo` input. This is the SHA-256 hash of the edge server's schema, represented as a hexadecimal string. 
+    * If this or any other `reportServerInfo` request fails with a non-2xx response from Apollo, the edge server should retry after 20 seconds.
+2. Apollo responds with a `ReportServerInfoResponse` object. This response tells the edge server: 
+    * How many seconds to wait before sending the next `reportServerInfo` request
+    * Whether the next `reportServerInfo` request should include the `executableSchema` that corresponds to the `executableSchemaId` provided in the previous request.
+3. After waiting the specified number of seconds, the edge server executes `reportServerInfo` again. This request includes a [normalized `executableSchema` string](#schema-normalization) if and only if `withExecutableSchema` was `true` in Apollo's most recent `ReportServerInfoResponse`.
+4. Go to step 2.
+
+### Pseudocode example
+
+```
+val info = EdgeServerInfo(..)
+val schema = normalize("type Query { .. }")
+var withSchema = false
+
+function sendReport() {
+  val executableSchema = if (withSchema) null else schema
+  val response = reportServerInfo(info, executableSchema)
+
+  if (response.withExecutableSchema) {
+    withSchema = true
+  }
+
+  setTimeout(sendReport, response.inSeconds)
+}
+
+sendReport()
+```
+
+## Type definitions
```

Comment:
This section needs an update. See https://github.com/mdg-private/monorepo/blob/master/docs/engine_services/auto_registration/auto_schema_reporting.md

Reply:
same for the step by step instructions above to match the types.